cmd/integration/harness.go

```diff
@@ -90,76 +77,28 @@ func NewHarness(cfg *ServerConfig, args []string) (*Harness, error) {
 	return h, nil
 }
 
-func isListening(addr string) bool {
-	conn, _ := net.DialTimeout("tcp", addr, 1*time.Second)
-	if conn != nil {
-		_ = conn.Close()
-		return true
-	}
-	return false
-}
-
-func killProcess(address string) error {
-	addr, err := net.ResolveTCPAddr("tcp", address)
-	if err != nil {
-		return err
-	}
-
-	var cmd *exec.Cmd
-	if runtime.GOOS == "windows" {
-		args := fmt.Sprintf("(Get-NetTCPConnection -LocalPort %d).OwningProcess -Force", addr.Port)
-		cmd = exec.Command("Stop-Process", "-Id", args)
-	} else {
-		args := fmt.Sprintf("lsof -i tcp:%d | grep LISTEN | awk '{print $2}' | xargs kill -9", addr.Port)
-		cmd = exec.Command("bash", "-c", args)
-	}
-
-	var errb bytes.Buffer
-	cmd.Stderr = &errb
-
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-
-	if err := cmd.Wait(); err != nil {
-		return fmt.Errorf("error during killing process: %s | %s", err, errb.String())
-	}
-
-	return nil
-}
-
 func main() {
+	log.JSONLog(true)
 	// os.Args[0] contains the current process path
 	h, err := NewHarnessDefaultServerConfig(os.Args[1:])
 	if err != nil {
-		log.Error("An error has occurred while generating a new harness: %v", err)
+		log.With().Error("An error has occurred while generating a new harness: ", log.Err(err))
 	}
 	// listen on error channel, quit when process stops
 	go func() {
 		for {
 			select {
 			case errMsg := <-h.server.errChan:
-				log.Error("harness received an err from subprocess: %v", errMsg)
+				log.With().Error("harness received an err from subprocess: ", log.Err(errMsg))
 			case <-h.server.quit:
 				log.Info("harness got quit signal from subprocess")
 				return
 			}
 		}
 	}()
 
-	// TODO: Expose server when it's needed to pass commands
-	// a dummy server so the main process won't be terminated before the tests are done running
-	//srv := &http.Server{Addr: ":9595"}
-	//defer func() {
-	//	if err := srv.Shutdown(context.TODO()); err != nil {
-	//		log.Error("cannot shutdown http server: ", err)
-	//	}
-	//}()
-	//
-	//err = srv.ListenAndServe()
-	//if err != nil {
-	//	log.Error("cannot start http server: ", err)
-	//}
+	log.With().Info("harness waiting for a quit signal from sub-process")
+	<-h.server.quit
```

Comment:
why not put this in the loop and remove the go routine?

Reply:
fixed in e5add110af94a4a0352e14bfa2befe99788470a0